aws/rust-runtime/aws-config/src/profile/parser/source.rs

```diff
@@ -184,7 +184,9 @@ fn expand_home(
 /// that we can check.
 fn check_is_likely_running_on_a_lambda(environment: &aws_types::os_shim_internal::Env) -> bool {
     // LAMBDA_TASK_ROOT – The path to your Lambda function code.
-    environment.get("LAMBDA_TASK_ROOT").is_ok()
+    // AWS_LAMBDA_RUNTIME_API - Double-check because LAMBDA_TASK_ROOT is redacted to any code running in Extensions
+    // https://docs.aws.amazon.com/lambda/latest/dg/runtimes-extensions-api.html
+    environment.get("LAMBDA_TASK_ROOT").is_ok() || environment.get("AWS_LAMBDA_RUNTIME_API").is_ok()
```

Comment:
From what I can tell in the docs, `AWS_LAMBDA_RUNTIME_API` is only available when using a custom runtime. Perhaps we should check `AWS_LAMBDA_FUNCTION_NAME` instead of either of these values. Do you know if `AWS_LAMBDA_FUNCTION_NAME` is set for extensions also?

Reply:
`AWS_LAMBDA_RUNTIME_API` is always set in a Lambda extension process, it is required for the communication between the extension and the runtime. 

I'll check on `AWS_LAMBDA_FUNCTION_NAME` and get back to you, if that works for both then we can reduce this back down to one check.

Reply:
Yep, `AWS_LAMBDA_FUNCTION_NAME` exists for both, so I've switched the check to use that.